### PR TITLE
Dsd 766/select bug fix

### PR DESCRIPTION
**Pull request by oliviawongnyc**
Fixes JIRA ticket xxx

## This PR does the following:

- Fixes bug described in [DSD-766](https://jira.nypl.org/browse/DSD-766).
- Adds snapshot testing for the `Select` component.

## How has this been tested?

- All previously existing and new tests pass.

## Accessibility concerns or updates

- No concerns or updates.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/Select/Select.test.tsx [hunk 1]
```diff
@@ -255,4 +256,100 @@ describe("Select", () => {
...
+        </Select>
+      )
+      .toJSON();
+    const disabled = renderer
+      .create(
+        <Select
+          id="select"
+          labelText="Which Succession sibling are you?"
+          name="succession-sibling"
+          isDisabled
```
Comment (EdwinGuzman) [inline src/components/Select/Select.test.tsx]: Nit: order the props alphabetically.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr874-c31r1hcb06nhetnxgnvr0toiwvkojwfq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620ea6ccdae2dd15a5c44164

File: src/components/Select/Select.test.tsx [hunk 2]
```diff
@@ -255,4 +256,100 @@ describe("Select", () => {
...
+    const primary = renderer
+      .create(
+        <Select
+          id="select"
+          labelText="Which Succession sibling are you?"
+          name="succession-sibling"
+        >
+          {siblings.map((sibling) => (
+            <option>{sibling}</option>
+          ))}
```
Comment (EdwinGuzman) [inline src/components/Select/Select.test.tsx]: This gets rendered often. You can define this in a variable and replace it in all the locations where this `.map` gets rendered.

**Review by EdwinGuzman — APPROVED** [review]
The comment about setting the rendered options in a variable is a nice-to-have optimization, but not something to prevent approving this PR.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr874-c31r1hcb06nhetnxgnvr0toiwvkojwfq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/620ea6ccdae2dd15a5c44164

File: src/components/Select/Select.test.tsx (hunk 2 above)
Comment (oliviawongnyc) [reply src/components/Select/Select.test.tsx]: > 

I did it!

**Pull request merged**